ui/apps/platform/src/Containers/Violations/ViolationsTablePage.tsx

```diff
@@ -114,22 +135,22 @@ function ViolationsTablePage(): ReactElement {
 
     // When any of the deps to this effect change, we want to reload the alerts and count.
     useEffect(() => {
-        const searchFilterWithViolationState = {
+        const enhancedSearchFilter: SearchFilter = {
             ...searchFilter,
+            ...getFilteredWorkflowSearchFilter(filteredWorkflowState),
             'Violation State': activeViolationStateTab,
         };
 
         const { request: alertRequest, cancel: cancelAlertRequest } = fetchAlerts(
-            searchFilterWithViolationState,
+            enhancedSearchFilter,
```

Comment:
Although not for change, I will add discussion item for our team meeting:
* Pro and con of separate arguments versus arguments object with named properties.
    For recent example,
    replace `getPaginationParams(page, perPage, sortOption)`
    with `getPaginationParams({ page, perPage, sortOption })`
* And then `searchFilter` from `useURLSearch` has become enhanced on more and more pages, so what is a mnemonic name for hypothetical `fetchAlerts({ mnemonicName, sortOption, page, perPage })`

Reply:
That's a good catch. I prefer using an argument object vs. separate arguments. My reason is as follows:
1. If we need to remove/add arguments, it's easier to do so with an argument object.
2. If some arguments are optional, managing with an object is easier.
3. The order of keys doesn't matter for an object.
4. It's easier to identify what the argument is supposed to represent when we use an argument object. Doing something like this
```
func("some value", true) 
```
isn't as clear as something like this
```
func({ value: "some value", isOn: true })
```
You can make it clearer by passing named variables, but the possibility of passing direct values can lead to confusion. 

And there are some cons too:
1. Harder to understand defaults with an object
2. An object might be overkill if a function has one or two arguments.

Reply:
But for this case, we should use an object. If a function has 3+ arguments, it probably makes sense to use an object.

Reply:
>  And then searchFilter from useURLSearch has become enhanced on more and more pages, so what is a mnemonic name for hypothetical fetchAlerts({ mnemonicName, sortOption, page, perPage })

@pedrottimark I get what you're saying. I'm not 100% sure what the best approach is here. I'm thinking maybe we could do something like this

```
const alertSearchFilter: SearchFilter = {
    ...searchFilter,
    ...getFilteredWorkflowSearchFilter(filteredWorkflowState),
     'Violation State': activeViolationStateTab,
};
```

Calling the whole object `alertSearchFilter` conveys that this search filter is specific to an alert. We take our base search filter and extend it. The `filteredWorkflowState` and `activeViolationStateTab` both are retrieved from the URL using a different hook. We convert it to the search filter format and add it to the `alertSearchFilter` object. We can convert `fetchAlerts` to take an argument object and rename `searchFilter` to `alertSearchFilter`.